### PR TITLE
Dotcom homepage prelim cleanup

### PR DESCRIPTION
**Pull request by st0nebraker**
This PR is paving some preliminary cleanup to dotcom's homepage, closing #43246.

Preliminary search guidance is added to an auth'd users `/search` view. (The current search "tips and tricks" section from unauth'd `/search`).

## Note
Because of the complexity in S2's search examples that don't quite work for open source searches & because suggested search section is [planning to get a revamp](https://www.figma.com/file/X2BEdyugA4AvZIAXC4iEuj?node-id=4:4305#290128068), we can re-use the open source search suggestion for both unauth'd & auth'd views in #43255.

## Test plan
- Ensure all tests pass
- Ensure visiting `/search` home page logged in on dotcom (locally `SOURCEGRAPHDOTCOM_MODE=true sg start web-standalone`) shows the same search query samples on k8s and s2
- Observe no breaking changes on logged out/in state in this view for all env's

## BEFORE
unauth'd:
image
auth'd:
image

## NOW
unauth'd:
image
auth'd:
image

## App preview:

- [Web](https://sg-web-becca-dotcom-homepage-cleanup.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-iyhnsrgmwu.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.25 kb) | 0.00% (+0.25 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/becca/dotcom-homepage-cleanup/compare-report.html;tab=live_object) for a full comparison between the commits c98f29d and 1a6010e or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/search/home/SearchPage.tsx [hunk 1]
```diff
@@ -78,6 +80,18 @@ export const SearchPage: React.FunctionComponent<React.PropsWithChildren<SearchP
                 })}
             >
                 {props.isSourcegraphDotCom && !props.authenticatedUser && <LoggedOutHomepage {...props} />}
+                {props.isSourcegraphDotCom && props.authenticatedUser && (
```
Comment (limitedmage) [inline client/web/src/search/home/SearchPage.tsx:83]: We should also check that `showEnterpriseHomePanels` is not enabled to show this.

**Review by limitedmage — APPROVED** [review]

**Comment by limitedmage** [comment]
Good thinking about using this existing component, this is a great workaround until we have full search suggestions on dotcom.

File: client/web/src/search/home/SearchPage.tsx (hunk 1 above)
Comment (st0nebraker) [reply client/web/src/search/home/SearchPage.tsx]: Right, good eye

File: client/web/src/search/home/SearchPage.tsx [hunk 2]
```diff
@@ -78,6 +80,18 @@ export const SearchPage: React.FunctionComponent<React.PropsWithChildren<SearchP
...
+                    <TipsAndTricks
+                        title="Tips and Tricks"
+                        examples={exampleTripsAndTricks}
+                        moreLink={{
+                            label: 'More search features',
+                            href: 'https://docs.sourcegraph.com/code_search/explanations/features',
+                            trackEventName: 'HomepageExampleMoreSearchFeaturesClicked',
```
[the remark this answers is not in the document]
Comment (st0nebraker) [reply client/web/src/search/home/SearchPage.tsx:90]: Hmm, we are going to change this in the next week or 2 to a different design & highlighted search queries. So maybe for now the data should stay how it is to not mess with the history, and we can add logging for logged in/logged out users with the next iteration. Are you cool with that?

[the remark this answers is not in the document]
Comment (limitedmage) [reply client/web/src/search/home/SearchPage.tsx:90]: Yeah that sounds good!

**Review by limitedmage — APPROVED** [review]

**Pull request merged**